integration-tests/tests/src/tests/results.ts

```diff
@@ -0,0 +1,1067 @@
+
+const PersonObjectSchema = {
+  name: "PersonObject",
+  properties: {
+    name: "string",
+    age: "double",
+    married: { type: "bool", default: false },
+    children: { type: "list", objectType: "PersonObject" },
+    parents: { type: "linkingObjects", objectType: "PersonObject", property: "children" },
+  },
+};
+
+class PersonObject extends Realm.Object {
+  name!: string;
+  age!: Realm.Types.Double;
+  married!: boolean;
+  children!: Realm.List<PersonObject>;
+  parents!: Realm.List<PersonObject>;
+}
+
+const TestObjectSchema = {
+  name: "TestObject",
+  properties: {
+    doubleCol: "double",
+  },
+};
+
+interface TestObject {
+  doubleCol: Realm.Types.Double;
+}
+
+const IntPrimarySchema = {
+  name: "IntPrimaryObject",
+  primaryKey: "primaryCol",
+  properties: {
+    primaryCol: "int",
+    valueCol: "string",
+  },
+};
+
+interface IntPrimaryObject {
+  primaryCol: Realm.Types.Int;
+  valueCol: string;
+}
+
+const DefaultValuesSchema = {
+  name: "DefaultValuesObject",
+  properties: {
+    boolCol: { type: "bool", default: true },
+    intCol: { type: "int", default: -1 },
+    floatCol: { type: "float", default: -1.1 },
+    doubleCol: { type: "double", default: -1.11 },
+    stringCol: { type: "string", default: "defaultString" },
+    dateCol: { type: "date", default: new Date(1.111) },
+    dataCol: { type: "data", default: new ArrayBuffer(1) },
+    objectCol: { type: "TestObject", default: { doubleCol: 1 } },
+    nullObjectCol: { type: "TestObject", default: null },
+    arrayCol: { type: "TestObject[]", default: [{ doubleCol: 2 }] },
+  },
+};
+
+const BasicTypesSchema = {
+  name: "BasicTypesObject",
+  properties: {
+    boolCol: "bool",
+    intCol: "int",
+    floatCol: "float",
+    doubleCol: "double",
+    stringCol: "string",
+    dateCol: "date",
+    dataCol: "data",
+    decimal128Col: "decimal128",
+    objectIdCol: "objectId",
+    uuidCol: "uuid",
+  },
+};
+
+interface BasicTypesObject {
+  boolCol: boolean;
+  intCol: Realm.Types.Int;
+  floatCol: Realm.Types.Float;
+  doubleCol: Realm.Types.Double;
+  stringCol: Realm.Types.String;
+  dateCol: Realm.Types.Date;
+  dataCol: Realm.Types.Data;
+  decimal128Col: Realm.Types.Decimal128;
+  objectIdCol: Realm.Types.ObjectId;
+  uuidCol: Realm.Types.UUID;
+}
+
+const LinkTypesSchema = {
+  name: "LinkTypesObject",
+  properties: {
+    objectCol: "TestObject",
+    objectCol1: { type: "object", objectType: "TestObject" },
+    arrayCol: "TestObject[]",
+    arrayCol1: { type: "list", objectType: "TestObject" },
+  },
+};
+
+const NullableBasicTypesSchema = {
+  name: "NullableBasicTypesObject",
+  properties: {
+    boolCol: "bool?",
+    intCol: "int?",
+    floatCol: "float?",
+    doubleCol: "double?",
+    stringCol: "string?",
+    dateCol: "date?",
+    dataCol: "data?",
+    decimal128Col: "decimal128?",
+    objectIdCol: "objectId?",
+    uuidCol: "uuid?",
+  },
+};
+
+describe("Results", () => {
+  beforeEach(Realm.clearTestState);
```

Comment:
This might not be needed, since it's default behaviour of `openRealmBeforeEach` to also clear test state after each test.

```suggestion
```